src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -683,18 +683,11 @@ internal static void CreateSymbolicLink(string path, string pathToTarget, bool i
             ValueStringBuilder sb = new(Interop.DefaultPathBufferSize);
             sb.Append(linkPath);
 
-            string? linkTarget = Interop.Sys.ReadLink(linkPath);
+            string? linkTarget = Interop.Sys.ReadLink(linkPath, isDirectory, true);
             if (linkTarget == null)
             {
                 sb.Dispose();
-                Interop.Error error = Interop.Sys.GetLastError();
-                // Not a link, return null
-                if (error == Interop.Error.EINVAL)
-                {
-                    return null;
-                }
-
-                throw Interop.GetExceptionForIoErrno(new Interop.ErrorInfo(error), linkPath, isDirectory);
```

Comment:
Thanks for working on this, but I don't understand the purpose of this change.  The only call site that ends up passing in true is the one that's already doing the error handling.  What is this fixing?

Reply:
I think the purpose is to not store ReadLink error for later use by callers since just one would profit from it.

Reply:
Maybe I did not understand the requirement well, should I do validation for all callers ?

Reply:
> I think the purpose is to not store ReadLink error for later use by callers all the the time since just one would profit from it.

We want to store the error.  Callers can then decide to look at it or not.  We don't want to bloat projects that don't need it.

> Maybe I did not understand the requirement well, should I do validation for all callers ?

The issue is to determine which call sites need to do better error handling and add whatever error handling is appropriate for those call sites.  ReadLink itself shouldn't need to be modified.

Reply:
At least this code needs to handle error codes: https://github.com/dotnet/runtime/blob/f5469fe3ca5fb5fb7f1c4a57ce6dac7c74e32d8c/src/libraries/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs#L43-L49 There's a few more cases that may need it as well, but I agree with @stephentoub, it should be done on the callsites.